doc/nrf/ug_bootloader_external_flash.rst

```diff
@@ -0,0 +1,67 @@
+.. _ug_bootloader_external_flash:
+
+Using external flash memory partitions
+######################################
+
+It is possible to use the external flash memory as the storage partition for the secondary slot.
+This requires a driver for the external flash memory that supports the following:
+
+* Single-byte read and write.
+* Writing data from the internal flash memory to the external flash memory.
+
+See :ref:`pm_external_flash` for general information about how to set up partitions in the external flash memory in the Partition Manager.
+
+.. note::
+
+   External flash memory partitions for image slots are currently only supported by MCUboot.
+
+To configure a bootloader to use the external flash memory for the secondary slot, edit the configuration file of the bootloader child image to specify an external flash region.
+For MCUboot, this configuration is the file :file:`ncs/bootloader/mcuboot/boot/zephyr/pm.yml`, and should have the region defined under ``mcuboot_secondary``:
+
+.. code-block:: yaml
+
+   mcuboot_secondary:
+       region: external_flash
+       size: CONFIG_PM_EXTERNAL_FLASH_SIZE
+
+The following example shows how to configure an application for the nRF52840 DK.
+The nRF52840 DK comes with the external flash memory that can be used for the secondary slot and that can be accessed using the QSPI NOR flash memory driver.
+
+1. Add the following configuration options to the :file:`ncs/bootloader/mcuboot/boot/zephyr/prj.conf` file:
+
+   .. code-block:: console
+
+      CONFIG_NORDIC_QSPI_NOR=y
+      CONFIG_NORDIC_QSPI_NOR_FLASH_LAYOUT_PAGE_SIZE=4096
+      CONFIG_NORDIC_QSPI_NOR_FLASH_ALLOW_STACK_USAGE_FOR_DATA_IN_FLASH=y
+      CONFIG_MULTITHREADING=y
+      CONFIG_BOOT_MAX_IMG_SECTORS=256
+      CONFIG_PM_EXTERNAL_FLASH=y
+      CONFIG_PM_EXTERNAL_FLASH_DEV_NAME="MX25R64"
+      CONFIG_PM_EXTERNAL_FLASH_SIZE=0xf4000
+      CONFIG_PM_EXTERNAL_FLASH_BASE=0
+
+   These options enable the QSPI NOR flash memory driver, multi-threading (which is required by the flash memory driver), and the external flash memory of the nRF52840 DK.
+
+#. Update the :file:`ncs/bootloader/mcuboot/boot/zephyr/pm.yml` file (as described above):
```

Comment:
Could we describe here how to create a file overriding settings in the specific sample directory? I don't think it is useful for anybody of us or our customers to modify the mcuboot project files.

Reply:
This is an issue with the pm.yml itself, we should add support in that file for enabling external flash support without modifying the pm.yml. In the future this will be replaced with what options to set in kconfig for this to work correctly.